pippy/PipelineStage.py

```diff
@@ -70,6 +70,10 @@ def __init__(
         self.kwargs_chunk_spec = kwargs_chunk_spec
         self.output_chunk_spec = output_chunk_spec
 
+        self.streams = []
+        for i in range(chunks):
+            self.streams.append(torch.cuda.Stream())
```

Comment:
this looks good to me, but originally I mean we can let users set how many streams they want to use.  We can let them set 2/3 or maybe like what you are doing here # of stages. Basically this number falls in the range of [2, # of stages]. But we can address this in a separate PR.

Reply:
resolved in a subsequent PR: https://github.com/pytorch/PiPPy/pull/852